### DISCUSSION TITLE
I want to add anti-disturbance function to the controller of crazyflie, what should I do?

### DISCUSSION THREAD

**Discussion by czyv587** [bitcraze/discussions#900] [Q&A]
I switched to the indi controller and flew under the wind disturbance of the electric fan. The flying effect was not as good as the PID controller, and the shaking was more violent than the PID controller. Is this the original effect of the indi controller or have I not adjusted any parameters properly?
As far as I know, there is an active disturbance rejection controller called Active Disturbance Rejection Controller(ADRC), but I don’t understand this algorithm very well. Can anyone give me some advice? Thank you so much!

**Comment by krichardsson** [comment]
The INDI controller was contributed to the code base and we (at Bitcraze) do not know that much about it. It is maybe best to try to get some information from the original contributors?
You can find the contributors by [searching pull requests (in the crazyflie firmware reop)  containing INDI](https://github.com/bitcraze/crazyflie-firmware/pulls?q=is%3Apr+is%3Aclosed+indi)
